_resources/port1.0/group/python-1.0.tcl

```diff
@@ -123,6 +123,11 @@ proc python_set_versions {option action args} {
             destroot {
                 system "echo $name is a stub port > ${destroot}${prefix}/share/doc/${name}/README"
             }
+            notes "
+            This is a stub port and does not install any binary files!
+            You will instead need to install one its subports!
+            You can safely uninstall this port.
```

Comment:
isn't the whole idea that the stub automatically installs the subport so the user specifically does not have to do this?

Reply:
No. I wish it were so. It only places a README file in share/docs with the contents stating that this is a stub port.

Reply:
`destroot` for the stub port only installs this:
`system "echo $name is a stub port > ${destroot}${prefix}/share/doc/${name}/README"`

And when trying to use the expected binaries that are not installed because it is a stub port, who would think to look into that README file to learn that this is a stub port.

I thought that displaying a note when a stub port is installed would be much more helpful to the user.

Reply:
Perhaps if MacPorts didn't even show stub ports when doing a `port search ...` would be better, OR if as you thought, the stub port matching the users default for `port select ...` for ruby/python/php/perl and if no default chosen, notify the user that they haven't set a default for ruby/python/php/perl and then give an error when attempting to install.

Reply:
I don't think any of these notes are really necessary, but if something gets added as notes then the message should probably be rephrased.
1. Python/Perl/Ruby ports typically do not install binary files anyway
2. `You will instead need to install one its subports!` should probably read `You need to install one of its subports instead!`

Reply:
>> isn't the whole idea that the stub automatically installs the subport so the user specifically does not have to do this?
>
> No. I wish it were so.

Ok, that weird behavior is unique to the ruby module ports, which have been a bit neglected. The perl, python and php module ports behave the way Ken described. I would rather someone fix the ruby module ports to work that way as well (by making a change to the ruby portgroup).

I guess the value to the user is that they are able to install the stub port and it will install the latest language version for them. If at some later time a newer version of the language becomes available, the user will learn about that when they upgrade the stub port. So they may keep the stub port installed to receive that benefit. Such a user probably does not want to see notes about their stub ports every time they're upgraded. So if anything, I would append a statement about stub ports to the long description rather than the notes.

Reply:
Ok. PortGroup ruby 1.0 now detects the default version or latest version supported by the ruby module and then depends on that supbort. (This implementation was taken from the perl 1.1 PortGroup.)

Reply:
Plus, notes are now removed and a note about the stub port is now appended to the long description (but this will not show up unless the `ruby.setup` etc. are after the long_description in the Portfile.